### DISCUSSION TITLE
I am trying to add seprate act boss music

### DISCUSSION THREAD

**Discussion by Raptorq1** [Techokami/SonicWorldsNext#73] [Q&A] [answered]
How can I add separate act boss music

**Comment by DimensionWarped** [comment]
You can just set Gobal.bossMusic to whichever music file you like. You might need to do some stuff with the stream afterwards... I'm not sure yet.

In the future, please tell us what effort you have made to figure out how to do something yourself when asking about it.

**Comment by DimensionWarped** [comment, marked as answer]
Alright, whipped this up real quick...

in Level.gd, add the following:

Add this near the other exports -
```
@export var bossMusic = preload("res://Audio/Soundtrack/5. SWD_Boss.ogg")
```

And Below...

```
	if Global.music != null:
		if music != null:
			Global.music.stream = music
			Global.music.play()
			Global.music.stream_paused = false
		else:
			Global.music.stop()
			Global.music.stream = null
```
	
Add this:

```
	if Global.bossMusic  != null:
		if bossMusic != null:
			Global.bossMusic.stream = bossMusic
```

Now you can change the boss music for the act in the same place as you set the normal music for the act.

Note that you can at any time change the boss music again by setting Global.bossMusic.stream, so if you need two different pieces of boss music to play in a single act that might be useful knowledge.

**Reply by Raptorq1** [reply]
It worked thanks for the help
